### DISCUSSION TITLE
Persisted State

### DISCUSSION THREAD

**Discussion by phuze** [beholdr/maska#195] [Q&A]
I am storing the maska object in a pinia store.

```vue
<script setup>
  import { vMaska } from "maska"
  import { formSore } from "@/stores/myForm"
  const store = formStore()
</script>

<template>
  <input
      type="text"
      name="phone"
      v-maska="store.form.phone"
      data-maska="(###) ###-####"
  />
</template>
```

My pinia store has persisted state enabled.

```js
import { defineStore } from "pinia"

export const formStore = defineStore({
  id: "myForm",
  persist: true,
  state: () => ({
    form: {
      first_name: '',
      last_name: '',
      email: '',
      phone: {},
    }
  })
})
```

Upon page reload, the maska input is not re-populated.
I can confirm that the object **is being persisted**.

How can I resolve this? See demonstration:

![maska-persisted-state](https://github.com/beholdr/maska/assets/9218035/8a378c2b-ff96-4176-b82a-70a7af26c2fb)

**Comment by beholdr** [comment]
Maybe you could use standard `v-model` binding for this case to save in the store?

**Comment by phuze** [comment]
**Update**:
@beholdr Could you please elaborate on what you mean?

If I utilize both `v-maska` and `v-model`, as demonstrated below, the field will be rendered on page reload as expected, **however**, maska will no longer generate the unmasked data as expected. Both `object.masked` and `object.unmasked` are the same value.

```vue
<script setup>
  import { vMaska } from "maska"
  import { formSore } from "@/stores/myForm"
  const store = formStore()
</script>

<template>
  <input
      type="text"
      name="phone"
      v-model="store.form.phone.unmasked"
      v-maska="store.form.phone"
      data-maska="(###) ###-####"
  />
</template>
```

```js
import { defineStore } from "pinia"

export const formStore = defineStore({
  id: "myForm",
  persist: true,
  state: () => ({
    form: {
      first_name: '',
      last_name: '',
      email: '',
      phone: {
        completed: '',
        masked: '',
        unmasked: ''
      },
    }
  })
})
```

![maska-persisted-state-error](https://github.com/beholdr/maska/assets/9218035/8016d7a5-d456-4df4-8dda-90674610c0b8)

**Reply by beholdr** [reply]
I mean you save in the store masked value as is, and if you need unmasked value, you can use Maska programmatically to get it before sending a form for example.
